crates/biome_js_analyze/src/lint/nursery/use_readonly_class_properties.rs

```diff
@@ -0,0 +1,517 @@
+                    return Some((
+                        AnyJsClassMemberName::JsPrivateClassMemberName(replace_class_member_name),
+                        leading,
+                    ));
+                }
+            }
+            None
+        }
+    }
+}
+
+impl AnyClassProperties {
+    pub fn text_trimmed(&self) -> Option<String> {
+        match self {
+            AnyClassProperties::JsPropertyClassMember(property_class_member) => {
+                let class_member_name = property_class_member.name();
+                if let Ok(class_member_name) = class_member_name {
+                    let member_name = class_member_name.name();
+                    if let Some(member_name) = member_name {
+                        let mut name = String::from(member_name.text());
+                        if class_member_name
+                            .as_js_private_class_member_name()
+                            .is_some()
+                        {
+                            name = format!("#{}", name);
+                        }
+                        return Some(name);
+                    }
+                }
+                None
+            }
+            AnyClassProperties::TsPropertyParameter(property_parameter) => {
+                match property_parameter.formal_parameter().ok()? {
+                    AnyJsFormalParameter::JsFormalParameter(formal_parameter) => Some(
+                        formal_parameter
+                            .binding()
+                            .ok()?
+                            .as_any_js_binding()?
+                            .as_js_identifier_binding()?
+                            .name_token()
+                            .ok()?
+                            .text_trimmed()
+                            .to_string(),
+                    ),
+                    _ => None,
+                }
+            }
+        }
+    }
+
+    pub fn range(&self) -> Option<TextRange> {
+        match self {
+            AnyClassProperties::JsPropertyClassMember(property_class_member) => {
+                Some(property_class_member.name().ok()?.range())
+            }
+            AnyClassProperties::TsPropertyParameter(property_parameter) => {
+                match property_parameter.formal_parameter().ok()? {
+                    AnyJsFormalParameter::JsBogusParameter(_)
+                    | AnyJsFormalParameter::JsMetavariable(_) => None,
+                    AnyJsFormalParameter::JsFormalParameter(param) => Some(
+                        param
+                            .binding()
+                            .ok()?
+                            .as_any_js_binding()?
+                            .as_js_identifier_binding()?
+                            .name_token()
+                            .ok()?
+                            .text_range(),
+                    ),
+                }
+            }
+        }
+    }
+
+    pub fn into_syntax(&self) -> SyntaxElement<JsLanguage> {
```

Comment:
Because you declared `AnyClassProperties` as an AST node union, `into_syntax()` or (`syntax().clone()`) should be available it. To get access to these methods, you need to import `AstNode`:

```
use biome_rowan::AstNode;
```